osutil/env.go

```diff
@@ -65,51 +68,192 @@ func GetenvInt64(key string, dflt ...int64) int64 {
 	return 0
 }
 
-// EnvMap takes a list of "key=value" strings and transforms them into
-// a map.
-func EnvMap(env []string) map[string]string {
-	out := make(map[string]string, len(env))
-	for _, kv := range env {
-		l := strings.SplitN(kv, "=", 2)
-		if len(l) == 2 {
-			out[l[0]] = l[1]
+// Environment is an unordered map of key=value strings.
+//
+// Environment can be manipulated with available methods and eventually
+// converted to low-level representation necessary when executing programs.
+// This approach discourages operations that could result in duplicate
+// environment variable definitions from being constructed.
+type Environment map[string]string
+
+func parseEnvEntry(entry string) (string, string, error) {
+	parts := strings.SplitN(entry, "=", 2)
+	if len(parts) != 2 {
+		return "", "", fmt.Errorf("cannot parse environment entry: %q", entry)
```

Comment:
Hmm, a
```
FOO="A=B"
```
is a valid env value, but we will fail here; I'm not sure how likely it is to encounter it in practice; should we try to be smarter here?

Reply:
Are you sure? Why would we fail? We expect exactly 2 items and use SplitN

Reply:
Yeah, you're right, I missed this. Sorry about the noise ;)